libraries/db/include/graphene/db/generic_index.hpp

```diff
@@ -67,10 +67,24 @@ namespace graphene { namespace chain {
 
          virtual void modify( const object& obj, const std::function<void(object&)>& m )override
          {
-            assert( nullptr != dynamic_cast<const ObjectType*>(&obj) );
-            auto ok = _indices.modify( _indices.iterator_to( static_cast<const ObjectType&>(obj) ),
-                                       [&m]( ObjectType& o ){ m(o); } );
-            FC_ASSERT( ok, "Could not modify object, most likely a index constraint was violated" );
+            assert(nullptr != dynamic_cast<const ObjectType*>(&obj));
+            bool exception = false;
+            auto ok = _indices.modify(_indices.iterator_to(static_cast<const ObjectType&>(obj)),
+                                       [&m, &exception](ObjectType& o) {
+                                          try {
+                                             m(o);
+                                          } catch (fc::exception e) {
+                                             exception = true;
+                                             elog("Exception while modifying object: ${e} -- object may be corrupted",
+                                                  ("e", e));
+                                          } catch (...) {
+                                             exception = true;
+                                             elog("Unknown exception while modifying object");
+                                          }
+                                       }
+                      );
+            FC_ASSERT(!exception, "Aborting for exception while modifying object");
```

Comment:
Best if can re-throw the real exception rather than covering it with a new message and logging it only, so a client program will know what happened.

Reply:
That is true. I went with the lazy option to start, because copying exceptions can be funny, but I'll play with it a bit more